### PR TITLE
boards/arm: Add arduino_header support to nucleo f429zi

### PR DESCRIPTION
**Pull request by chrta**
Add arduino_header support to nucleo f429zi board.

Signed-off-by: Christian Taedcke <hacking@taedcke.com>

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Would you mind moving this into a new arduino_r3_connector.dtsi file (along with other arduino declarations)?
I'm trying to enforce this model to relieve board.dts

**Comment by chrta** [comment]
I moved the Arduino stuff in a separate file and used a separate commit for that. Or should I squash it?

File: boards/arm/nucleo_f429zi/nucleo_f429zi.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ toolchain:
 ram: 192
 flash: 2048
 supported:
+  - arduino_gpio
```
Comment (erwango) [inline boards/arm/nucleo_f429zi/nucleo_f429zi.yaml]: Please add arduino_spi as well

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks, minor addition needed.
And squashing would be nice indeed, thanks!

File: boards/arm/nucleo_f429zi/nucleo_f429zi.yaml (hunk 1 above)
Comment (chrta) [reply boards/arm/nucleo_f429zi/nucleo_f429zi.yaml]: done

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**